File: BE_QuanLyTro/src/main/java/ut/edu/be_quanlytro/Dto/Request/ContractMemberAddRequest.java
package ut.edu.be_quanlytro.Dto.Request;

import lombok.Getter;
import lombok.Setter;

import java.time.LocalDate;
import java.util.UUID;

@Getter @Setter
public class ContractMemberAddRequest {
    private UUID contractId;
    private String fullName;
    private String phone;
    private LocalDate dob;
    private String hometown;
    private String idCardNumber;
}

File: Flutter_QuanLyTro/lib/data/models/request/contract_member_add_request.dart
class ContractMemberAddRequest {
  final String contractId;
  final String fullName;
  final String phone;
  final String dob; // Định dạng 'yyyy-MM-dd'
  final String hometown;
  final String idCardNumber;

  ContractMemberAddRequest({
    required this.contractId,
    required this.fullName,
    required this.phone,
    required this.dob,
    required this.hometown,
    required this.idCardNumber,
  });

  Map<String, dynamic> toJson() {
    return {
      'contractId': contractId,
      'fullName': fullName,
      'phone': phone,
      'dob': dob,
      'hometown': hometown,
      'idCardNumber': idCardNumber,
    };
  }
}